cumulus_etl/http.py

```diff
@@ -0,0 +1,213 @@
+"""HTTP helper methods"""
```

Comment:
This file is mostly moved code. Here are all the changes I believe:
- tweaked how authentication is caught - brought it out of the old equivalent of `_request_once` and put it into the retry loop - it felt more honest to only actually make one request in `_request_once`.
- Look for a few more possible error messages inside an error reponse payload (this lets us simplify some of the oauth2 error handling code and just have it be done here)
- Catch httpx transport errors (like DNS issues) and tranform them into a errors.FatalNetworkError exception, so that the rest of the code only needs to catch one kind of error.